src/transformers/configuration_utils.py

```diff
@@ -275,7 +275,7 @@ def from_dict(cls, config_dict: Dict, **kwargs) -> "PretrainedConfig":
         """
         return_unused_kwargs = kwargs.pop("return_unused_kwargs", False)
 
-        config = cls(**config_dict)
+        config = cls(**config_dict, **kwargs)
```

Comment:
Isn't that what line 287 below is supposed to do... ?

Reply:
Indeed, this probably isn't the best way to do things, however, it currently fails because during initialisation we initialize the `id2label` and `label2id` according to `num_labels`:

```py
        self.id2label = kwargs.pop("id2label", {i: "LABEL_{}".format(i) for i in range(self.num_labels)})
        self.id2label = dict((int(key), value) for key, value in self.id2label.items())
        self.label2id = kwargs.pop("label2id", dict(zip(self.id2label.values(), self.id2label.keys())))
        self.label2id = dict((key, int(value)) for key, value in self.label2id.items())
```

Since we update it afterwards, it doesn't update the `id2label` and `label2id` values. The best way would probably be to use a setter?